### PR TITLE
Update dependencies and minor improvements

### LINKED ISSUE #50: Failure to download binaries should give other exit code than 0
If sonarqube-scanner command is unable to download sonar-scanner it will print an error messagen and then exit with status code 0. It would be better if it could exit with another status code.
In comparison, if there is a connection problem to the sonar-qube server, the program will exit with code 1.

```
npx sonarqube-scanner
...
[15:37:34] Trying to find a local install of the SonarScanner: C:\Users\xxxx\.sonar\native-sonar-scanner\sonar-scanner-4.7.0.2747-windows\bin\sonar-scanner.bat
Path not found.
[15:37:34] Proceed with download of the platform binaries for SonarScanner...
[15:37:34] Creating C:\Users\xxxx\.sonar\native-sonar-scanner
[15:37:34] error in downloader
[15:37:34] Error: getaddrinfo ENOTFOUND binaries.sonarsource.com
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26) {
  errno: -3008,
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'binaries.sonarsource.com'
}
[15:37:34] ERROR: impossible to download and extract binary: getaddrinfo ENOTFOUND binaries.sonarsource.com
[15:37:34]        SonarScanner binaries probably don't exist for your OS (windows).
[15:37:34]        In such situation, the best solution is to install the standard SonarScanner (requires a JVM).
[15:37:34]        Check it out at https://redirect.sonarsource.com/doc/install-configure-scanner.html

$ echo $?
0
```

It would be great if this behaviour could be changed, since failing quietly most often means that the uses have to spend more time troubleshooting.

### PR DESCRIPTION
**Pull request by victor-diez-sonarsource**
Also fixes #50

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,23 +25,23 @@
...
+    "mocha": "^10.2.0",
     "nyc": "^15.1.0",
-    "prettier": "^2.8.0",
-    "rimraf": "^3.0.2",
-    "sinon": "^15.0.0",
-    "typescript": "^4.9.5"
+    "prettier": "^3.0.3",
+    "rimraf": "^5.0.1",
+    "sinon": "^15.2.0",
+    "typescript": "<5.2"
```
Comment (yassin-kammoun-sonarsource) [inline package.json:44]: Is there a reason why you force TypeScript below 5.2?

File: src/bin/sonar-scanner [hunk 2]
```diff
@@ -1,9 +1,13 @@
 #!/usr/bin/env node
-var scan = require('../index').cli
+var scan = require('../index').scan;
```
Comment (yassin-kammoun-sonarsource) [inline src/bin/sonar-scanner:2]:
```suggestion
const scan = require('../index').scan;
```

File: src/bin/sonar-scanner [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
 
-var options = []
+var options = [];
```
Comment (yassin-kammoun-sonarsource) [inline src/bin/sonar-scanner:4]:
```suggestion
let options = [];
```

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
A minor question to address. Looks good otherwise!

File: package.json (hunk 1 above)
Comment (victor-diez-sonarsource) [reply package.json:44]: because with typescript-eslint <6 it displays a message that typescript should be <5.2

File: src/bin/sonar-scanner (hunk 2 above)
Comment (victor-diez-sonarsource) [reply src/bin/sonar-scanner:2]: ok, didn't change any var, not sure if we are supposed to support old js. I guess given that we require node >14 we can change them all

File: package.json (hunk 1 above)
Comment (yassin-kammoun-sonarsource) [reply package.json:44]: I didn't realize that we were using TypeScript ESLint in this repo. Thanks for the clarification!

**Pull request merged**